test/e2e/apimachinery/namespace.go

```diff
@@ -357,9 +356,9 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
 			return err
 		})
 		framework.ExpectNoError(err, "failed to update namespace status %s", ns)
-		framework.ExpectEqual(len(statusUpdated.Status.Conditions), len(statusUpdated.Status.Conditions), fmt.Sprintf("updated object should have the applied condition, got %#v", statusUpdated.Status.Conditions))
-		framework.ExpectEqual(string(statusUpdated.Status.Conditions[len(statusUpdated.Status.Conditions)-1].Type), "StatusUpdate", fmt.Sprintf("updated object should have the approved condition, got %#v", statusUpdated.Status.Conditions))
```

Comment:
The typecast to string was dropped here.

Reply:
how about it?

```
gomega.Expect(statusUpdated.Status.Conditions[len(statusUpdated.Status.Conditions)-1].Type).To(gomega.Equal(v1.NamespaceConditionType("StatusUpdate")), "updated object should have the approved condition, got %#v", statusUpdated.Status.Conditions)
```

Reply:
I raise a pr to fix it. https://github.com/kubernetes/kubernetes/pull/120479